app/assets/javascripts/miq_application.js

```diff
@@ -1403,6 +1403,11 @@ function miqToolbarOnClick(e) {
         }
       }
     }
+  } else if (button.data('function') {
+    // hack to support data-function and data-function-data
+    var fn = new Function(button.data('function')); // eval
```

Comment:
This is actually broken because `new Function("foobar")` returns a function that evaluates `foobar` and ..nothing, doesn't call it, whereas `new Function("foobar()")` would call it but can't pass in arguments...

Simple fix:

```js
var fn = new Function("return " + button.data('function')); // eval - returns a function returning the right function
fn().call(button, button.data('functionData'));
```

And then any function name (without parentheses) will work in `data-function`.